api/krusty/options.go

```diff
@@ -18,6 +18,10 @@ type Options struct {
 	// order as specified by the kustomization file(s).
 	DoLegacyResourceSort bool
 
+	// When true, use full-depth git clone, in order to allow
+	// reference to a specific commit
+	DoDeepGitClone bool
```

Comment:
instead a bool, please make this an enum defined in the types package ala types.LoadRestrictions

suggest `GitBehavior`, with values `DeepClone`, `SpeedyClone` (or something better)

This allows more behaviors in the future w/o adding more booleans.  Also, Options is part of the high level exposed API, so want it to be clear as possible, and want to avoid adding bool arguments to call stack.  DoLegacyResourceSort is exceptional (in a bad way), so left it as a boolean defined so that false worked as a default.